### PR TITLE
camera rotation follow cursor

### PR DESCRIPTION
**Pull request by celsodantas**
ping @ericsobral @vinioliveira

A raw version (not using SLERP) to make the camera follow the cursor. Here is the proof:

screen shot 2014-06-14 at 8 09 56 pm

This implementation is not suuuuper smooth. And maybe too fast. It's hard to have complete control of the rotation. Maybe I should look for how to do this using [SLERP](http://en.wikipedia.org/wiki/Slerp).

Another thing: I added a `tick` method to the camera. All game objects will have to have that method and will be called every frame update (so it can update itself). 

Currently I'm passing the `engine` to the object butI should pass different object that would have all data needed (like camera, buttons pressed, environment data, etc) for all game objects. I'll change this once I need it.

I decided not to make mouse movements events. Maybe I'll not make any input as an event. For now, I decided to check for inputs state during each frame. I think it make things less crazy and each object can do it's own check. (ex: check if the `jump-button` was pressed)

The event system will still exist but not for inputs. I think.

Thoughts?

### CODE REVIEW

**Comment by vinioliveira** [comment]
As a row version I think is good enough. Just the cohesion and the coupling I think could be better. 

Keep going :+1:

**Comment by celsodantas** [comment]
thanks! As a tip, add comments to the lines in the PR and no the commit. =)

**Comment by vinioliveira** [comment]
Yep, this was a general comment about the PR that's why it's posted here.

On Sunday, June 15, 2014, Celso Dantas notifications@github.com wrote:

> thanks! As a tip, add comments to the lines in the PR and no the commit. =)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/celsodantas/topaz/pull/2#issuecomment-46121597.

## 

Marcus Vinícius Oliveira
@vinimsoliveira

**Comment by celsodantas** [comment]
I'm talking about the other comments, not the last one. You made comments to the lines of the commits, not the lines in the PR.

**Comment by vinioliveira** [comment]
I see ! Thanks for the tip :)

On Sunday, June 15, 2014, Celso Dantas notifications@github.com wrote:

> I'm talking about the other comments, not the last one. You made comments
> to the lines of the commits, not the lines in the PR.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/celsodantas/topaz/pull/2#issuecomment-46121732.

## 

Marcus Vinícius Oliveira
@vinimsoliveira

File: src/Game.cpp [hunk 1]
```diff
@@ -16,11 +16,14 @@ Topaz::Game::Game(Engine::Core engine) : _engine(engine) {
...
 
 void Topaz::Game::run() {
-  // This should
   _running = true;
 
   while (_running)
   {
+    _engine.updateCursorData();
+    _engine.updateDeltaData();
+    _engine.camera.tick(_engine);
```
Comment (celsodantas) [inline src/Game.cpp:25]: I'll move all this to inside the `engine` class

**Pull request merged**